components/logging/org.wso2.carbon.logging.service/src/main/java/org/wso2/carbon/logging/service/util/LoggingUtil.java

```diff
@@ -429,6 +431,21 @@ public static int getLineNumbers(String logFile) throws Exception {
         return logsList.toArray(new String[logsList.size()]);
     }
 
+    public static void restoreDefaults() throws Exception {
+        registryManager.removeAllRegistryEntries();
+        LogManager.resetConfiguration();
+
+        try {
+            String logFile = CarbonUtils.getCarbonConfigDirPath()
+                    + RegistryConstants.PATH_SEPARATOR + "log4j.properties";
+            Log4jConfigurer.initLogging(logFile);
+        } catch (FileNotFoundException e) {
+            String msg = "Cannot restore default logging configuration."
+                    + " log4j.properties file not found in the classpath";
+            throw new Exception(msg, e);
```

Comment:
Yes, you are correct. It is always better to throw a specific exception. In this case, LogViewerException can be thrown since this is related to logging viewer.

Fixed in,
https://github.com/wso2/carbon-commons/pull/265